programs/clearing_house/src/state/oracle_map.rs

```diff
@@ -61,9 +64,53 @@ impl<'a> OracleMap<'a> {
         Ok(self.price_data.get(pubkey).unwrap())
     }
 
+    pub fn get_price_data_and_validity(
+        &mut self,
+        pubkey: &Pubkey,
+        last_oracle_price_twap: i128,
+    ) -> ClearingHouseResult<(&OraclePriceData, OracleValidity)> {
+        if pubkey == &Pubkey::default() {
+            return Ok((&self.quote_asset_price_data, OracleValidity::Valid));
+        }
+
+        if self.price_data.contains_key(pubkey) {
+            let oracle_price_data = self.price_data.get(pubkey).unwrap();
+            let is_oracle_valid = oracle_validity(
+                last_oracle_price_twap,
+                oracle_price_data,
+                &self.oracle_guard_rails.validity,
+            )?;
+            return Ok((oracle_price_data, is_oracle_valid));
+        }
+
+        let (account_info, oracle_source) = match self.oracles.get(pubkey) {
+            Some(AccountInfoAndOracleSource {
+                account_info,
+                oracle_source,
+            }) => (account_info, oracle_source),
+            None => {
```

Comment:
might want to add a msg! which says what the pubkey of the not found oracle is for easier debugging